command/base.go

```diff
@@ -106,7 +107,13 @@ func (c *BaseCommand) Client() (*api.Client, error) {
 		config.Address = c.flagAddress
 	}
 	if c.flagAgentProxyAddress != "" {
-		config.Address = c.flagAgentProxyAddress
+		config.AgentAddress = c.flagAgentProxyAddress
+	}
+	if c.flagVaultProxyAddress != "" {
+		err := config.SetTransportProxy(c.flagVaultProxyAddress)
+		if err != nil {
+			return nil, fmt.Errorf("failed to set Vault proxy address: %w", err)
```

Comment:
Re: my comment in the thread, this should be updated to not refer to Vault Proxy. I was mistaken about `VAULT_PROXY_ADDR`